crates/client/rpc-core/src/lib.rs

```diff
@@ -145,4 +146,25 @@ pub trait StarknetRpcApi {
     /// contract.
     #[method(name = "getProof")]
     fn get_proof(&self, get_proof_input: RpcGetProofInput) -> RpcResult<RpcGetProofOutput>;
+
+    /// Simulate a given sequence of transactions on the requested state, and generate the execution
+    /// traces. If one of the transactions is reverted, raises error.
+    /// Spec version 0.4.0-rc2
+    #[method(name = "simulateTransactions")]
+    fn simulate_transactions(
+        &self,
+        block_id: BlockId,
+        transactions: Vec<BroadcastedTransaction>,
+        simulation_flags: Vec<SimulateTransactionFlag>,
+    ) -> RpcResult<Vec<SimulateTransactionResult>>;
+
+    /// Compatibility with Starknet.js
+    /// See https://github.com/0xs34n/starknet.js/blob/829f5f4375a63c7c269e34dad22c9cd8874a64ba/src/provider/rpc.ts#L546
+    #[method(name = "simulateTransaction")]
```

Comment:
wdym ? this endpoint isn't in the specs ?

Reply:
In 0.3.0 it was `simulateTransaction` with `transaction` field
In 0.4.0 it is `simulateTransactions` with `transactions` field
In starknet.js it is `simulateTransaction` with `transactions` field